src/main/scala/com/databricks/spark/csv/CsvRelation.scala

```diff
@@ -53,7 +53,12 @@ case class CsvRelation protected[spark] (
   extends BaseRelation with TableScan with PrunedScan with InsertableRelation {
 
   // Share date format object as it is expensive to parse date pattern.
-  private val dateFormatter = if (dateFormat != null) new SimpleDateFormat(dateFormat) else null
+  private val dateFormatter: Seq[SimpleDateFormat] =
+    if (dateFormat != null) {
+      dateFormat.split(",").map(pattern => new SimpleDateFormat(pattern))
```

Comment:
I suggest supporting up to three additional options: `dateFormat1`, `dateFormat2`, `dateFormat3` (including the old `dateFormat`) and do not try to tokenize the string passed by user.